uaclient/entitlements/base.py

```diff
@@ -211,22 +211,28 @@ def is_access_expired(self):
 
     def process_contract_deltas(
             self, orig_access: 'Dict[str, Any]',
-            deltas: 'Dict[str, Any]') -> bool:
+            deltas: 'Dict[str, Any]', allow_enable: bool = False) -> bool:
         """Process any contract access deltas for this entitlement.
 
         :param orig_access: Dictionary containing the original
             resourceEntitlement access details.
         :param deltas: Dictionary which contains only the changed access keys
         and values.
+        :param allow_enable: Boolean set True if allowed to perform the enable
+            operation. When False, a message will be logged to inform the user
+            about the recommended enabled service.
 
         :return: True when delta operations are processed; False when noop.
         """
         if not deltas:
             return True  # We processed all deltas that needed processing
-        transition_to_unentitled = False
-        if orig_access and 'entitled' in deltas.get('entitlement', {}):
-            transition_to_unentitled = (
-                deltas['entitlement']['entitled'] in (False, util.DROPPED_KEY))
+
+        delta_entitlement = deltas.get('entitlement', {})
+        transition_to_unentitled = bool(delta_entitlement == util.DROPPED_KEY)
```

Comment:
No need for the `bool` call here; `==` is going to return a `bool` (unless someone does something _very_ weird in a custom `__eq__` method).